h2o-py/h2o/utils/shared_utils.py

```diff
@@ -353,6 +360,129 @@ def slice_is_normalized(s):
+    :param genmodelpath: This is the local file system folder name where both h2o-genmodel.jar and MOJO zip file is stored.
+    :param labels:     (Optional) True : Shows results, False: Does no show results and just pass result to given object
+    :param javaoptions: (Optional) These are the Java string options given by the user
+    :param show_debug: True/False - Default ( Use True to see the full Java command line)
+    :return: score as json values
+    """
+    # Verifying JSON
+    if show_debug:
+        print(json)
+
+    check_json(json)
+
+    # Checking java
+    java = H2OLocalServer._find_java()
+
+    jver_bytes = subprocess.check_output([java, "-version"], stderr=subprocess.STDOUT)
+    jver = jver_bytes.decode(encoding="utf-8", errors="ignore")
+    if show_debug:
+        print("  Java Version: " + jver.strip().replace("\n", "; "))
+    if "GNU libgcj" in jver:
+        raise H2OStartupError("Sorry, GNU Java is not supported for H2O.\n"
+                              "Please download the latest 64-bit Java SE JDK from Oracle.")
+    if "Client VM" in jver:
+        warn("  You have a 32-bit version of Java. H2O works best with 64-bit Java.\n"
+             "  Please download the latest 64-bit Java SE JDK from Oracle.\n")
+
+    # genmodelpath > must start and ends with "/" or add to it
+    gen_model_arg = ".:"
+
+    ## Working on mojo_model
+    mojo_model_path=None
+    head, tail = ntpath.split(mojo_model)
+    if not head:
+        if not genmodelpath:
+            raise RuntimeError("Error: given mojo model " + mojo_model + " does not have full path!")
+        else:
+            mojo_model_path = genmodelpath
+    else:
+        mojo_model_path = head
+
+    if not mojo_model_path:
+        raise RuntimeError("Error: given mojo model " + mojo_model + " does not have full path!")
+
+    # mojo model name is
+    mojo_model_name = tail
+
+
+    # Mojo model is given by name only and separate path is given also
+    # Check both genmodel and pojo is available in the same path
+    mojo_test = find_file(mojo_model_name, mojo_model_path)
+    if mojo_test is None:
+        raise RuntimeError("Error: MOJO model " + mojo_model_name + " is not available in " + mojo_model_path + " folder.")
+
+
+    genmodel_test = find_file(gen_model_file_name, mojo_model_path)
+    if genmodel_test is None:
+        raise RuntimeError("Error:" + gen_model_file_name +  " is not available in " + mojo_model_path + " folder.")
+
+
+    temp_dir_path =  mojo_model_path
+
+    if mojo_model_path.endswith("/"):
+        gen_model_arg = gen_model_arg + mojo_model_path
+        temp_dir_path = mojo_model_path[:-1]
+    else:
+        gen_model_arg = gen_model_arg + mojo_model_path + "/"
+
+    gen_model_arg = (gen_model_arg
+                     + gen_model_file_name + ":"
+                     + temp_dir_path
+                     + ":genmodel.jar:/")
+
+    if (show_debug):
+        print(gen_model_arg)
+
+    if mojo_model_path.endswith("/"):
```

Comment:
use existing library functions to create the path

Reply:
it is not creating the path, it is just checking if the path ends with "/" or not because in the genmodelpath user can give a path with "/" at the end or not.

Reply:
I was referring to the whole block of code. Lines 467-470 are composing a path. This can be substituted for a library call (os.path.join).

Example:
```
>>> os.path.join("/a", "b")
'/a/b'
>>> os.path.join("/a/", "b")
'/a/b'
```